core/core-api/src/main/java/com/hbc/pms/core/api/service/PlcService.java

```diff
@@ -0,0 +1,85 @@
+package com.hbc.pms.core.api.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.hbc.pms.core.api.constants.StationEnum;
+import com.hbc.pms.core.api.support.json.Coordinate;
+import com.hbc.pms.core.api.support.json.PlcCoordinates;
+import com.hbc.pms.plc.integration.mokka7.S7Client;
+import com.hbc.pms.plc.integration.mokka7.exception.S7Exception;
+import com.hbc.pms.plc.integration.mokka7.type.AreaType;
+import com.hbc.pms.plc.integration.mokka7.type.DataType;
+import com.hbc.pms.plc.integration.mokka7.util.S7;
+import jakarta.annotation.PostConstruct;
+import lombok.Getter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+@Service
+public class PlcService {
+
+  private final Logger log = LoggerFactory.getLogger(getClass());
+  private final S7Client plcClient = new S7Client();
+
+  @Getter
+  public final Map<String, PlcCoordinates> plcCoordinatesOfStations = new HashMap<>();
+
+  @Value("${hbc.plc.url}")
+  private String plcUrl;
+
+  @PostConstruct
+  public void postConstruct() {
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+      PlcCoordinates plcCoordinates;
+      for (StationEnum stationName : StationEnum.values()) {
+        plcCoordinates = mapper.readValue(
+            new ClassPathResource(String.format("plc-coordinate/%s.json", stationName.getName())).getInputStream(),
+            PlcCoordinates.class
+        );
+        plcCoordinatesOfStations.put(stationName.getName(), plcCoordinates);
+        log.info(mapper.writeValueAsString(plcCoordinates));
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    // TODO: add retries in case the PLC connection is interrupted
+    try {
+      plcClient.connect(plcUrl, 0, 1);
+    } catch (S7Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public boolean readBoolean(Coordinate coordinate) throws S7Exception {
+    validateType(coordinate, "boolean");
+    return plcClient.readBit(AreaType.DB, coordinate.getDb(), coordinate.getStartByte(), 0);
+  }
+
+  public int readInt(Coordinate coordinate) throws S7Exception {
+    validateType(coordinate, "int");
+    return plcClient.readInt(AreaType.DB, coordinate.getDb(), coordinate.getStartByte());
+  }
+
+  public float readFloat(Coordinate coordinate) throws S7Exception {
+    validateType(coordinate, "float");
+    final byte[] buffer = new byte[1024];
+    plcClient.readArea(AreaType.DB, coordinate.getDb(), coordinate.getStartByte(), 1, DataType.REAL, buffer);
+    return S7.getFloatAt(buffer, 0);
```

Comment:
In my opinion, you can move it to the library. And please double check [**real** type in PLC](https://docs.plm.automation.siemens.com/content/plant_sim_help/15/plant_sim_all_in_one_html/en_US/tecnomatix_plant_simulation_help/simtalk_reference/simtalk_reference_1/data_types/real_data_type.html) matches with float or double.

Reply:
According to the document you provided, the **real** data type of PLC should be matched with the **double** data type of Java.

Reply:
@zKirinz please add a method to library if real type in PLC matches with double type in Java